packages/template-retail-react-app/app/contexts/customer-product-lists.js

```diff
@@ -0,0 +1,61 @@
+/*
+ * Copyright (c) 2022, Salesforce, Inc.
+ * All rights reserved.
+ * SPDX-License-Identifier: BSD-3-Clause
+ * For full license text, see the LICENSE file in the repo root or https://opensource.org/licenses/BSD-3-Clause
+ */
+
+import React, {useContext} from 'react'
+import {useCurrentCustomer} from '../hooks/use-current-customer'
+import PropTypes from 'prop-types'
+import {CustomerProductListsContext} from './index'
+import {useCustomerProductLists, useShopperCustomersMutation} from 'commerce-sdk-react-preview'
+
+export const CustomerProductListsProvider = ({children}) => {
+    const {data: customer} = useCurrentCustomer()
+    const {isRegistered, customerId} = customer
+    const createCustomerProductList = useShopperCustomersMutation('createCustomerProductList')
+    const {data: productLists, ...restOfQuery} = useCustomerProductLists(
+        {
+            parameters: {customerId}
+        },
+        {
+            onError: (e) => {
+                console.error('e', e)
+            },
+            onSuccess: (data) => {
+                if (!data.total) {
+                    createCustomerProductList.mutate({
+                        parameters: {customerId},
+                        // we only use one type of product list for now
+                        body: {type: 'wish_list'}
+                    })
+                }
+            },
+            // only registered user can have product lists
+            enabled: isRegistered
+        }
+    )
+    const value = {
+        productLists,
+        ...restOfQuery
+    }
```

Comment:
Here, the `productLists` is mixed with the rest of the query state and method. The Provider has value look something like this

<img width="322" alt="Screenshot 2023-03-03 at 2 58 17 PM" src="https://user-images.githubusercontent.com/10948652/222850138-34e777ee-2416-4367-9873-ce0605e68ec6.png">

Is this intentional to duplicate `data` and rename it to `productLists`? Can we just use `data`?

Reply:
I think we have the same problem for `useCurrentBasket`, if we have to build abstractions, it must follow the underlying existing patterns. For example, if the abstraction is built on top of react-query, it should always return the same type as react-query. Altering the type introduces new APIs that are unfamiliar to devs.